CVE-2024-5932: WordPress GiveWP Plugin RCE (#19424)
rapid7/metasploit-framework: documentation/modules/exploit/multi/http/wp_givewp_rce.md

## Vulnerable Application

This Metasploit module exploits an unauthenticated PHP Object Injection vulnerability in the
GiveWP plugin for WordPress (versions <= 3.14.1).
The vulnerability is present in the 'give_title' parameter, allowing attackers to inject a crafted

> PHP object that can lead to remote code execution (RCE) when combined with a suitable POP chain.

Reviewer: ```suggestion
PHP object leading to remote code execution (RCE) when combined with a suitable POP chain.
```

## Setup

> 1. **Docker Compose Setup**: Use the following `docker-compose.yml` file to set up a vulnerable WordPress environment.

Reviewer 2: ```suggestion
1. **Docker Compose Setup**: Create the following `docker-compose.yml` file to set up a vulnerable WordPress environment:
```

```yaml
services:
  db:
    image: mysql:8.0.27
    command: '--default-authentication-plugin=mysql_native_password'
    restart: always
    environment:
      - MYSQL_ROOT_PASSWORD=somewordpress
      - MYSQL_DATABASE=wordpress
      - MYSQL_USER=wordpress
      - MYSQL_PASSWORD=wordpress
    expose:
      - 3306
      - 33060

  wordpress:
    image: wordpress:6.3.2
    ports:
      - "80:80"
    restart: always
    environment:
      - WORDPRESS_DB_HOST=db
      - WORDPRESS_DB_USER=wordpress
      - WORDPRESS_DB_PASSWORD=wordpress
      - WORDPRESS_DB_NAME=wordpress
volumes:
  db_data:
```

> 

Reviewer 2: ```suggestion
1. Run Docker: `docker compose up`
1. Access the WordPress instance at `http://127.0.0.1` and complete the installation process
```

> 2. **Download and Install Vulnerable GiveWP Plugin**:

Reviewer 2: ```suggestion
1. **Download and Install Vulnerable GiveWP Plugin**:
```
